doc/devguide/codebase/code-style.rst

```diff
@@ -6,12 +6,165 @@ Suricata uses a fairly strict coding style. This document describes it.
+"""""""""""""""""""""""
+
+.. Argh, github does not support admonitions such as .. note::
+
++--------------------------------------------------------------------+
+| **Note**                                                           |
+|                                                                    |
+| Do not reformat whole files by default, i.e. do not use            |
+| clang-format proper in general.                                    |
++--------------------------------------------------------------------+
+
+If you were ever to do so, formatting changes of existing code with clang-format
+shall be a different commit and must not be mixed with actual code changes.
+
+.. code-block:: bash
+
+    $ clang-format -i {file}
+
+Disabling clang-format
+**********************
+
+There might be times, where the clang-format's formatting might not please.
+This might mostly happen with macros, arrays (single or multi-dimensional ones),
+struct initialization, or where one manually formatted code.
+
+You can always disable clang-format.
+
+.. code-block:: c
+
+    /* clang-format off */
+    #define APP_LAYER_INCOMPLETE(c, n) (AppLayerResult){1, (c), (n)}
+    /* clang-format on */
+
+Installing clang-format and git-clang-format
+********************************************
+clang-format 9 is required.
+
+TODO - REMOVEME: We could possibly provide an "install-clang-format" script if we wanted to.
+It's essentially a decision of maintaining the install script vs support questions about
+"hey I have ubuntu xyz, what do I need to do?".
+
+On ubuntu 18.04:
+
+- It is sufficient to only install clang-format, e.g.
+
+    .. code-block:: bash
+
+        $ wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        $ sudo add-apt-repository "deb http://apt.llvm.org/bionic/   llvm-toolchain-bionic-9  main"
+        $ sudo apt-get update
+        $ sudo apt-get install -y clang-format-9
+
+- see http://apt.llvm.org for other releases
+
+On fedora:
+
+- Install the ``clang``  and ``git-clang-format`` packages with
+
+    .. code-block:: bash
+
+        $ sudo dnf install clang git-clang-format
+
+
 Line length
 ^^^^^^^^^^^
 
-There is a soft limit of ~80 characters.
+There is a soft limit of ~80 characters. There is a hard limit of 100.
+
+When wrapping lines that are too long, they should be indented at least 8
+spaces from previous line. You should attempt to wrap the minimal portion of
+the line to meet the 80 character limit.
+
+TODO - REMOVEME: 80 vs 100 width? @victorjulien mentioned maybe longer?
+Even linux is on 100 these days. However, keeping it to 80 would induce the
+fewest changes if existing code gets reformatted.
+
+TODO - REMOVEME: We should remove the "soft limit" as clang-format only has a
+hard limit.
+
+TODO - REMOVEME: Reflow comments, i.e. also adjust comments to length? Yup, for
+@jasonish so I set it to do so.
+
+clang-format:
+ - ColumnLimit: 80
+ - ContinuationIndentWidth: 8
+ - ReflowComments: true
```

Comment:
@jasonish I've enabled reformatting of comments. Also in the actual .clang-format file